includes/cache/SquidUpdate.php

```diff
@@ -122,13 +122,33 @@ static function purge( $urlArr ) {
 			return;
 		}
 
-		global $wgPurgeSquidViaCelery;
+		global $wgPurgeSquidViaCelery, $wgPurgeVignetteUsingSurrogateKeys;
 		if ( $wgPurgeSquidViaCelery == true ) {
-			( new AsyncCeleryTask() )
-					->taskType('celery_workers.purger.purge')
-					->setArgs( $urlArr, [] )
-					->setPriority( PurgeQueue::NAME )
-					->queue();
+			if ( $wgPurgeVignetteUsingSurrogateKeys == true ) {
+				// Filter array for vignette urls and split them into separate task
+				$vignetteUrls = [];
+				foreach ( $urlArr as $index => $url ) {
+					if ( VignetteRequest::isVignetteUrl($url) ) {
+						$vignetteUrls[] = $url;
+						unset($urlArr[$index]);
```

Comment:
Minor: Another way to do this might look like:

```PHP
$sets = array_reduce($urlArr, function($carry, $item) {
   if ( VignetteRequest::isVignetteUrl($url) ) {
      $carry['vignette'][] = $url;
   } else {
      $carry['other'][] = $url;
   }
}, array('vignette' => array(), 'other' => array());
```

Then

```php
if ( !empty($sets['vignette']) ) {
...
}
```